securitycenter/assets/list_all_assets.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Pacakge assets contains example snippets for listing CSCC Assets.
+package assets
+
+// [START list_all_assets]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	securitycenter "cloud.google.com/go/securitycenter/apiv1"
+	"google.golang.org/api/iterator"
+	securitycenterpb "google.golang.org/genproto/googleapis/cloud/securitycenter/v1"
+)
+
+// listAllAssets prints every asset  to w for orgID and returns the number of
+// assets found.  orgID is the numeric Organization ID.
+func listAllAssets(w io.Writer, orgID string) (int, error) {
+	// orgID := "12321311"
+	// Instantiate a context and a security service client to make API calls.
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return -1, fmt.Errorf("Error instantiating client %v\n", err)
+	}
+	defer client.Close() // Closing the client safely cleans up background resources.
+
+	req := &securitycenterpb.ListAssetsRequest{
+		Parent: fmt.Sprintf("organizations/%s", orgID),
+	}
+
+	assetsFound := 0
+	it := client.ListAssets(ctx, req)
+	for {
+		result, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return -1, fmt.Errorf("Error listing assets: %v", err)
+		}
+		asset := &result.Asset
+		properties := &(*asset).SecurityCenterProperties
+		fmt.Fprintf(w, "Asset Name: %s, Resource Name %s, Resource Type %s\n",
+			(*asset).Name,
+			(*properties).ResourceName,
+			(*properties).ResourceType)
```

Comment:
The standard guidance is to not break in the middle of function calls. See https://github.com/golang/go/wiki/CodeReviewComments#line-length.

Sometimes, when a sample is embedded in documentation, it can be hard to read if the line is too long. So, you might want to break this into multiple print statements? It will help a little bit once the pointer stuff is removed.